arch/arc/core/cache.c

```diff
@@ -125,28 +406,156 @@ int arch_dcache_invd_range(void *start_addr_ptr, size_t size)
 		start_addr += line_size;
 	} while (start_addr < end_addr);
 	irq_unlock(key); /* -exit critical section- */
+}
+
+#endif /* CONFIG_DCACHE_REGION_OPERATIONS */
+
+int arch_dcache_flush_range(void *start_addr_ptr, size_t size)
+{
+	size_t line_size = sys_cache_data_line_size_get();
+
+	if (!dcache_available() || (size == 0U) || line_size == 0U) {
+		return -ENOTSUP;
+	}
+
+#if defined(CONFIG_DCACHE_REGION_OPERATIONS)
+	dcache_flush_region(start_addr_ptr, size);
+#else
+	dcache_flush_lines(start_addr_ptr, size);
+#endif
+
+#if defined(CONFIG_SLC)
+	slc_flush_region(start_addr_ptr, size);
+#endif
+
+	return 0;
+}
+
+int arch_dcache_invd_range(void *start_addr_ptr, size_t size)
+{
+	size_t line_size = sys_cache_data_line_size_get();
+
+	if (!dcache_available() || (size == 0U) || line_size == 0U) {
+		return -ENOTSUP;
+	}
+
+#if defined(CONFIG_DCACHE_REGION_OPERATIONS)
+	dcache_invalidate_region(start_addr_ptr, size);
+#else
+	dcache_invalidate_lines(start_addr_ptr, size);
+#endif
+
+#if defined(CONFIG_SLC)
+	slc_invalidate_region(start_addr_ptr, size);
+#endif
 
 	return 0;
 }
 
 int arch_dcache_flush_and_invd_range(void *start_addr_ptr, size_t size)
 {
+#if defined(CONFIG_DCACHE_REGION_OPERATIONS)
```

Comment:
This function follows a different flow/structure. I assume it is because there was no (line based) implementation before. ok to leave it like this, but maybe for consistency you could add the line-based variant as well?